s3/src/main/java/ch/cyberduck/core/sts/AssumeRoleWithWebIdentitySTSCredentialsConfigurator.java

```diff
@@ -0,0 +1,108 @@
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSSessionCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.AnonymousAWSCredentials;
+import com.amazonaws.auth.profile.internal.AllProfiles;
+import com.amazonaws.auth.profile.internal.BasicProfile;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+import com.amazonaws.services.securitytoken.model.AssumeRoleWithWebIdentityRequest;
+import com.amazonaws.services.securitytoken.model.AssumeRoleWithWebIdentityResult;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class AssumeRoleWithWebIdentitySTSCredentialsConfigurator extends AWSProfileSTSCredentialsConfigurator {
+    private static final Logger log = LogManager.getLogger(AssumeRoleWithWebIdentitySTSCredentialsConfigurator.class);
+
+    public AssumeRoleWithWebIdentitySTSCredentialsConfigurator(final X509TrustManager trust, final X509KeyManager key, final PasswordCallback prompt) {
+        super(trust, key, prompt);
+    }
+
+    @Override
+    public Credentials configure(final Host host) {
+        final Credentials credentials = new Credentials(host.getCredentials());
+
+        final AWSSecurityTokenService service = this.getTokenService(host, null, null, null, null);
+
+        AssumeRoleWithWebIdentityRequest webIdReq = new AssumeRoleWithWebIdentityRequest()
+                .withWebIdentityToken(credentials.getOauth().getAccessToken());
+//                .withPolicy("policy")
+//                .withRoleArn("consoleAdmin")
+//                .withRoleSessionName("testSession");
+//                .putCustomQueryParameter("Version", "2011-06-15");
+
+        if(log.isDebugEnabled()) {
+            log.debug("Assuming role with web identity for host: {}", host);
+        }
+
+        AssumeRoleWithWebIdentityResult result = service.assumeRoleWithWebIdentity(webIdReq);
+        com.amazonaws.services.securitytoken.model.Credentials cred = result.getCredentials();
+
+        if(log.isDebugEnabled()) {
+            log.debug(cred.toString());
+        }
+
+        credentials.setUsername(cred.getAccessKeyId());
+        credentials.setPassword(cred.getSecretAccessKey());
+        credentials.setToken(cred.getSessionToken());
+
+        return credentials;
+    }
+
+    @Override
+    public AWSSecurityTokenService getTokenService(final Host host, final String region, final String accessKey, final String secretKey, final String sessionToken) {
+        return AWSSecurityTokenServiceClientBuilder
+                .standard()
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(String.format("%s://%s:%s", host.getProtocol().getScheme(), host.getHostname(),
```

Comment:
Most certainly a different STS endpoint than the bookmark configuration. Must be configurable in connection profile.